### PR TITLE
cleanup usage of __super with base declaration

### PR DESCRIPTION
**Pull request by pollend**
I was considering trying to get a couple of these utilities working on linux. I don't think the __super extension is available for g++ so I think its a good early step to resolve these compiler errors. I prefer the more explicit calls, it's a lot easier to understand without trying to figure out how this squares with multi inheritance.

https://gcc.gnu.org/legacy-ml/gcc/2004-08/msg01276.html

### CODE REVIEW

File: prog/tools/sharedInclude/EditorCore/ec_object.h [hunk 1]
```diff
@@ -16,14 +16,14 @@
 ///     @b false in other case\n
 /// virtual DClassID @b getClassId () - Returns class ID
 /// @ingroup EditorCore
-#define EO_IMPLEMENT_RTTI(id)                    \
+#define EO_IMPLEMENT_RTTI(id, base)              \
```
Comment (pollend) [inline prog/tools/sharedInclude/EditorCore/ec_object.h:19]: there is another version of this macro in ec_object.h

**Comment by NicSavichev** [comment]
> I was considering trying to get a couple of these utilities working on linux. I don't think the __super extension is available for g++ so I think its a good early step to resolve these compiler errors. I prefer the more explicit calls, it's a lot easier to understand without trying to figure out how this squares with multi inheritance.

I'm doing the same at the moment (expect shaderCompiler and dabuild to be buildabale and usable under linux/macOS).
But daEditorX and AssetViewer are not expected to be ported, they are too heavy windows-related.

**Comment by pollend** [comment]
> > I was considering trying to get a couple of these utilities working on linux. I don't think the __super extension is available for g++ so I think its a good early step to resolve these compiler errors. I prefer the more explicit calls, it's a lot easier to understand without trying to figure out how this squares with multi inheritance.
> 
> I'm doing the same at the moment (expect shaderCompiler and dabuild to be buildabale and usable under linux/macOS).
> But daEditorX and AssetViewer are not expected to be ported, they are too heavy windows-related.

Umm I see, anything in particular. I don't mind addressing minor upkeep to just get familiar.

File: prog/tools/sharedInclude/EditorCore/ec_object.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (pollend) [reply prog/tools/sharedInclude/EditorCore/ec_object.h:19]: why that doesn't seem to be what this macro is for? that would be a hidden side effect of using this, its not consistently used in the code base to begin with.

**Comment by NicSavichev** [comment]
Change is not relevant anymore (is refactored other way)

**Pull request closed without merging**